cibuildwheel/platforms/android.py

```diff
@@ -0,0 +1,429 @@
+
+
+def android_triplet(identifier: str) -> str:
+    return {
+        "arm64_v8a": "aarch64-linux-android",
+        "x86_64": "x86_64-linux-android",
+    }[parse_identifier(identifier)[1]]
+
+
+def parse_identifier(identifier: str) -> tuple[str, str]:
+    match = re.fullmatch(r"cp(\d)(\d+)-android_(.+)", identifier)
+    if not match:
+        msg = f"invalid Android identifier: '{identifier}'"
+        raise ValueError(msg)
+    major, minor, arch = match.groups()
+    return (f"{major}.{minor}", arch)
+
+
+@dataclass(frozen=True)
+class PythonConfiguration:
+    version: str
+    identifier: str
+    url: str
+
+    @property
+    def arch(self) -> str:
+        return parse_identifier(self.identifier)[1]
+
+
+def all_python_configurations() -> list[PythonConfiguration]:
+    return [PythonConfiguration(**item) for item in resources.read_python_configs("android")]
+
+
+def get_python_configurations(
+    build_selector: BuildSelector, architectures: set[Architecture]
+) -> list[PythonConfiguration]:
+    return [
+        c
+        for c in all_python_configurations()
+        if c.arch in architectures and build_selector(c.identifier)
+    ]
+
+
+def shell_prepared(command: str, build_options: BuildOptions, env: dict[str, str]) -> None:
+    shell(
+        prepare_command(command, project=".", package=build_options.package_dir),
+        env=env,
+    )
+
+
+def before_all(options: Options, python_configurations: list[PythonConfiguration]) -> None:
+    before_all_options = options.build_options(python_configurations[0].identifier)
+    if before_all_options.before_all:
+        log.step("Running before_all...")
+        shell_prepared(
+            before_all_options.before_all,
+            before_all_options,
+            before_all_options.environment.as_dictionary(os.environ),
+        )
+
+
+@dataclass
+class Builder:
+    config: PythonConfiguration
+    build_options: BuildOptions
+    tmp_dir: Path
+    built_wheels: list[Path]
+
+    def build(self) -> None:
+        log.build_start(self.config.identifier)
+        self.tmp_dir.mkdir()
+        self.setup_python()
+
+        compatible_wheel = find_compatible_wheel(self.built_wheels, self.config.identifier)
+        if compatible_wheel:
+            print(
+                f"\nFound previously built wheel {compatible_wheel.name} that is "
+                f"compatible with {self.config.identifier}. Skipping build step..."
+            )
+            repaired_wheel = compatible_wheel
+        else:
+            self.setup_env()
+            self.before_build()
+            repaired_wheel = self.build_wheel()
+
+        self.test_wheel(repaired_wheel)
+        if compatible_wheel is None:
+            self.built_wheels.append(
+                move_file(repaired_wheel, self.build_options.output_dir / repaired_wheel.name)
+            )
+
+        shutil.rmtree(self.tmp_dir)
+        log.build_end()
+
+    def setup_python(self) -> None:
```

Comment:
Flagging that this should probably differentiate `setup_target_python()` from `setup_build_python()` if a dependency on standalone is added.